ui/components/CodeEditor/CodeEditor.js

```diff
@@ -0,0 +1,55 @@
+import React from "react";
+import AceEditor from "react-ace";
+
+import "ace-builds/src-noconflict/mode-yaml";
+import "ace-builds/src-noconflict/theme-twilight";
+import yaml from "js-yaml";
+
+// eslint-disable-next-line react/prop-types
+const CodeEditor = ({ defaultValue = "", name = "YAML Editor" }) => {
+  const [value, setValue] = React.useState(defaultValue);
+  const [annotations, setAnnotations] = React.useState([]);
+  const checkerFunction = (value) => {
+    try {
+      console.log(yaml.load(value));
+    } catch (error) {
+      console.log(error.mark.line, error.mark.column, error.message, error.name);
+      setAnnotations((annotations) => [
+        {
+          row: error.mark.line,
+          column: error.mark.column,
+          text: error.message,
+          type: error.name,
+        },
+        ...annotations,
+      ]);
+    }
+  };
+  return (
+    <div>
+      <AceEditor
+        mode="yaml"
+        theme="twilight"
+        name={name}
+        onChange={(value) => {
+          setValue(value);
+          checkerFunction(value);
+        }}
+        editorProps={{
+          $blockScrolling: true,
+        }}
+        fontSize={12}
+        value={value}
+        setOptions={{
+          useWorker: false,

```

Comment:
interesting 🤔 , we might spin up a web worker if we have to parse huge files.

Reply:
I had to set that fix some errors I think. I don't remember properly. But yeah, I think we can look into it later when we need to parse big files.

Reply:
Yep